src/storage/src/hummock/state_store.rs

```diff
@@ -58,18 +61,27 @@ impl HummockStorage {
         validate_epoch(pinned_version.safe_epoch(), epoch)?;
 
         // check epoch if lower mce
-        let read_version_tuple = if epoch <= pinned_version.max_committed_epoch() {
-            // read committed_version directly without build snapshot
-            (Vec::default(), Vec::default(), (**pinned_version).clone())
-        } else {
-            // TODO: use read_version_mapping for batch query
-            let read_version_vec = vec![self.read_version.clone()];
-            let key_range = (
-                Bound::Included(TableKey(key.to_vec())),
-                Bound::Included(TableKey(key.to_vec())),
-            );
-            read_filter_for_batch(epoch, table_id, &key_range, read_version_vec)?
-        };
+        let read_version_tuple: (Vec<ImmutableMemtable>, Vec<SstableInfo>, CommittedVersion) =
+            if epoch <= pinned_version.max_committed_epoch() {
+                // read committed_version directly without build snapshot
+                (Vec::default(), Vec::default(), (**pinned_version).clone())
+            } else {
+                let read_version_vec = {
+                    let read_guard = self.read_version_mapping.read();
+                    read_guard
+                        .get(&table_id)
+                        .unwrap()
```

Comment:
We may return a default value if the table_id is not found.

Reply:
I think table_id not found is unacceptable